Mathlib/Probability/Distributions/Beta.lean

```diff
@@ -0,0 +1,213 @@
+    filter_upwards with a (_ : a > _)
+    have hx_a : ¬(0 < a ∧ a < 1) := by
+      simp only [not_and_or, not_lt]
+      right
+      linarith
+
+    rw [if_neg hx_a]
+
+/-- The beta pdf is measurable. -/
+@[measurability]
+lemma measurable_betaPDFReal (a b : ℝ) : Measurable (betaPDFReal a b) :=
+  Measurable.ite
+    measurableSet_Ioo (((measurable_id'.pow_const _).const_mul _).mul
+    ((measurable_const.sub' measurable_id').pow_const _))
+    measurable_const
+
+/-- The beta pdf is strongly measurable -/
+@[measurability]
+ lemma stronglyMeasurable_betaPDFReal (a b : ℝ) :
+     StronglyMeasurable (betaPDFReal a b) :=
+   (measurable_betaPDFReal a b).stronglyMeasurable
+
+open Measure
+
+/-- A Lebesgue Integral from -∞ to y can be expressed as the sum of one from -∞ to 0 and 0 to x -/
+lemma lintegral_Iio_eq_lintegral_Iic_add_Ioo {y z : ℝ} (f : ℝ → ℝ≥0∞) (hzy : z < y) :
+    ∫⁻ x in Iio y, f x = (∫⁻ x in Iic z, f x) + ∫⁻ x in Ioo z y, f x := by
+  rw [← Iic_union_Ioo_eq_Iio hzy, lintegral_union measurableSet_Ioo]
+  rw [Set.disjoint_iff]
+  rintro x ⟨h1 : x ≤ _, h2, _⟩
+  linarith
+
+/-- The pdf of the beta distribution integrates to 1 -/
+@[simp]
+lemma lintegral_betaPDF_eq_one {a b : ℝ} (ha : 0 < a) (hb: 0 < b) :
+    ∫⁻ x, betaPDF a b x = 1 := by
+  have left : ∫⁻ x in Iic 0, betaPDF a b x = 0 := by
+    rw [setLIntegral_congr_fun measurableSet_Iic
+      (ae_of_all _ (fun x (hx : x ≤ 0) ↦ betaPDF_of_non01 (Or.inl hx))), lintegral_zero]
+
+  have middle : ∫⁻ x in Ioo 0 1, betaPDF a b x =
+      ∫⁻ x in Ioo 0 1, ENNReal.ofReal (1 / (Beta a b) * x ^ (a - 1) * (1 - x) ^ (b - 1)) :=
+    setLIntegral_congr_fun measurableSet_Ioo (ae_of_all _ (fun _ ↦ betaPDF_of_01))
+
+  have right : ∫⁻ x in Ici 1, betaPDF a b x = 0 := by
+    rw [setLIntegral_congr_fun measurableSet_Ici
+      (ae_of_all _ (fun x (hx : x ≥ 1) ↦ betaPDF_of_non01 (Or.inr hx))), lintegral_zero]
+
+  have leftmid : ∫⁻ x in Iio 1, betaPDF a b x =
+  ∫⁻ x in Ioo 0 1, ENNReal.ofReal (1 / (Beta a b) * x ^ (a - 1) * (1 - x) ^ (b - 1)) := by
+    have expand : ∫⁻ x in Iio 1, betaPDF a b x =
+    (∫⁻ x in Iic 0, betaPDF a b x) + (∫⁻ x in Ioo 0 1, betaPDF a b x) := by
+      apply lintegral_Iio_eq_lintegral_Iic_add_Ioo (betaPDF a b)
+      apply zero_lt_one
+    rw [expand, left, middle, zero_add]
+
+  rw [← lintegral_add_compl _ measurableSet_Ici, compl_Ici,
+    leftmid, right, zero_add]
+  rw [← ENNReal.toReal_eq_one_iff, ← integral_eq_lintegral_of_nonneg_ae]
+
+  · simp_rw [mul_assoc]
+    rw [integral_mul_left]
+    rw [← BetaIntegral_ofReal ha hb]
+    ring_nf
+    apply DivisionRing.mul_inv_cancel
+    have h_betapos := Beta_pos ha hb
+    positivity
+  · rw [EventuallyLE, ae_restrict_iff' measurableSet_Ioo]
+    exact ae_of_all _ (fun x (hx : 0 < x ∧ x < 1) ↦ by
+      rw [Pi.zero_apply]
+      have hpos := betaPDFReal_nonneg ha hb
+      unfold betaPDFReal at hpos
+      specialize hpos x
+      rw [if_pos hx] at hpos
+      exact hpos
+    )
```

Comment:
Here is a bit shorter way, feel free to add it or not. However, please avoid to write multiline tactic blocks inside the arguments of a theorem. Use `apply` or `refine` instead.
```suggestion
    refine ae_of_all _ fun x hx ↦ ?_
    convert betaPDFReal_nonneg ha hb x using 1
    rw [betaPDFReal, if_pos (mem_Ioo.1 hx)]
```

Reply:
ah got it